polaris-react/src/components/Icon/Icon.tsx

```diff
@@ -76,12 +76,16 @@ export function Icon({source, color, backdrop, accessibilityLabel}: IconProps) {
   );
 
   const SourceComponent = source;
+  const iconName = (source as React.FunctionComponent)?.name
+    ?.replace('Svg', '')
+    .replace(/\d+$/, '');
```

Comment:
the component name is `SvgFlagMajor` for example or `SvgFlagMajor2`. This removes the `Svg` and any trailing numbers so the result will be `FlagMajor`